EIPS/eip-4762.md

```diff
@@ -89,25 +89,50 @@ Where `tree_key` and `sub_key` are computed as `tree_key, sub_key = get_storage_
+
+* Example 1: for a `CODECOPY` with start position 100, read size 50, `code_size = 200`, `x = 100` and `y = 149`
+* Example 2: for a `CODECOPY` with start position 600, read size 0, no chunks are accessed
+* Example 3: for a `CODECOPY` with start position 1500, read size 2000, `code_size = 3100`, `x = 1500` and `y = 3099`
+
+ If `EXTCODECOPY` targets an EOFv1 container, no code chunks are accessed since the code has hardcoded value `EF00`:
+
+Note that `CODESIZE`, `EXTCODESIZE` and `EXTCODEHASH` do NOT access any chunks.
 
- * At each step of EVM execution, if and only if `PC < len(code)`, chunk `PC // CHUNK_SIZE` (where `PC` is the current program counter) of the callee is accessed. In particular, note the following corner cases:
-     * The destination of a `JUMP` (or positively evaluated `JUMPI`) is considered to be accessed, even if the destination is not a jumpdest or is inside pushdata
-     * The destination of a `JUMPI` is not considered to be accessed if the jump conditional is `false`.
-     * The destination of a jump is not considered to be accessed if the execution gets to the jump opcode but does not have enough gas to pay for the gas cost of executing the `JUMP` opcode (including chunk access cost if the `JUMP` is the first opcode in a not-yet-accessed chunk)
-     * The destination of a jump is not considered to be accessed if it is beyond the code (`destination >= len(code)`)
-     * If code stops execution by walking past the end of the code, `PC = len(code)` is not considered to be accessed
- * If the current step of EVM execution is a `PUSH{n}`, all chunks `(PC // CHUNK_SIZE) <= chunk_index <= ((PC + n) // CHUNK_SIZE)` of the callee are accessed.
- * If a nonzero-read-size `CODECOPY` or `EXTCODECOPY` read bytes `x...y` inclusive, all chunks `(x // CHUNK_SIZE) <= chunk_index <= (min(y, code_size - 1) // CHUNK_SIZE)` of the accessed contract are accessed.
-     * Example 1: for a `CODECOPY` with start position 100, read size 50, `code_size = 200`, `x = 100` and `y = 149`
-     * Example 2: for a `CODECOPY` with start position 600, read size 0, no chunks are accessed
-     * Example 3: for a `CODECOPY` with start position 1500, read size 2000, `code_size = 3100`, `x = 1500` and `y = 3099`
- * `CODESIZE`, `EXTCODESIZE` and `EXTCODEHASH` do NOT access any chunks.
-    When a contract is created, access chunks `0 ... (len(code)+30)//31`
+When a contract is created, access all corresponding chunks `0 ... (cod_size+30)//31`, where `code_size` is `len(code)` in legacy contracts, or the container size in EOFv1 accounts.
+
+#### Access events for EOFv1 container
+
+If a `*CALL`, `EXT*CALL` instruction targets an account with the EOFv1 container format or a `EOFCREATE` instruction is executed, the code chunks of the following bytecodes are considered accessed:
+
+* The entire `header` section of the current (`*CALL`/`EXT*CALL`) or targeted (`EOFCREATE`) (sub-)container.
```

Comment:
`EOFCREATE` requires the entire target (sub-)container in order to hash it (used to calculate the new address, in the same vein as `CREATE2`), so should consider it accessed.

However, this particular detail has recently been disputed. I'll track this and in case the initcontainer hashing disappears, we'll circle back here and ask for an update.